src/sentry/utils/sdk.py

```diff
@@ -455,20 +453,97 @@ def _kwargs_into_scope(self, scope, extra=None, tags=None, fingerprint=None, req
+        # First check that the tag exists, because though it's true that "no value yet" doesn't
+        # match "some new value," we don't want to flag that as a mismatch.
+        if tag_key in scope._tags and scope._tags[tag_key] != expected_value:
+            scope.set_tag("possible_mistag", True)
+            scope.set_tag(f"scope_bleed.{tag_key}", True)
             extra = {
-                f"previous_{tag_key}": scope._tags[tag_key],
-                f"new_{tag_key}": expected_value,
+                f"previous_{tag_key}_tag": scope._tags[tag_key],
+                f"new_{tag_key}_tag": expected_value,
             }
+            merge_context_into_scope("scope_bleed", extra, scope)
             logger.warning(f"Tag already set and different ({tag_key}).", extra=extra)
-            return True
+
+
+def get_transaction_name_from_request(request: Request) -> str:
+    """
+    Given an incoming request, derive a parameterized transaction name, if possible. Based on the
+    implementation in `_set_transaction_name_and_source` in the SDK, which is what it uses to label
+    request transactions. See https://github.com/getsentry/sentry-python/blob/6c68cf4742e6f65da431210085ee095ba6535cee/sentry_sdk/integrations/django/__init__.py#L333.
+
+    If parameterization isn't possible, use the request's path.
+    """
+
+    transaction_name = request.path_info
+    try:
+        # Note: In spite of the name, the legacy resolver is still what's used in the python SDK
+        transaction_name = LEGACY_RESOLVER.resolve(
+            request.path_info, urlconf=getattr(request, "urlconf", None)
+        )
+    except Exception:
+        pass
+
+    return transaction_name
+
+
+def check_current_scope_transaction(
+    request: Request,
+) -> dict[str, str] | None:
+    """
+    Check whether the name of the transaction on the current scope matches what we'd expect, given
+    the request being handled.
+
+    If the transaction values match, return None. If they don't, return a dictionary including both
+    values.
+
+    Note: Ignores scope `transaction` values with `source = "custom"`, indicating a value which has
+    been set maunually. (See the `transaction_start` decorator, for example.)
+    """
+
+    with configure_scope() as scope:
+        transaction_from_request = get_transaction_name_from_request(request)
+
+        if (
+            scope._transaction != transaction_from_request
+            and scope._transaction_info.get("source") != "custom"
+        ):
+            return {
+                "scope_transaction": scope._transaction,
+                "request_transaction": transaction_from_request,
+            }
+
+
+def capture_exception(
+    error, scope: Scope | None = None, request: Request | None = None, **scope_args
+):
+    """
+    A wrapper around `sentry_sdk.capture_exception` which checks scope `transaction` against the
+    given Request object, to help debug scope bleed problems.
+    """
```

Comment:
I couldn't decide if it was better to let this shadow the SDK's `capture_exception` method (which until now this module has just been re-exporting) or to give it a different, more explicit, name (`capture_exception_with_scope_check` or something).

I went with the former (under the theory that it will require smaller changes in the spots where it's used), but could easily be talked into the latter.